File: app/src/main/java/com/guliash/countryquiz/core/BaseActivity.java
package com.guliash.countryquiz.core;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.guliash.countryquiz.utils.LifecycleLogger;

public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        LifecycleLogger.create(this);
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onDestroy() {
        LifecycleLogger.destroy(this);
        super.onDestroy();
    }

    @Override
    protected void onStart() {
        LifecycleLogger.start(this);
        super.onStart();
    }

    @Override
    protected void onStop() {
        LifecycleLogger.stop(this);
        super.onStop();
    }

    @Override
    protected void onResume() {
        LifecycleLogger.resume(this);
        super.onResume();
    }

    @Override
    protected void onPause() {
        LifecycleLogger.pause(this);
        super.onPause();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        LifecycleLogger.save(this);
        super.onSaveInstanceState(outState);
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
        LifecycleLogger.restore(this);
        super.onRestoreInstanceState(savedInstanceState);
    }

    @Override
    protected void onNewIntent(Intent intent) {
        LifecycleLogger.newIntent(this);
        super.onNewIntent(intent);
    }
}

File: app/src/main/java/com/guliash/quizzes/core/ui/activity/BaseActivity.kt
package com.guliash.quizzes.core.ui.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

open class BaseActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        println("${this}: onCreate")
        super.onCreate(savedInstanceState)
    }

    override fun onDestroy() {
        println("${this}: onDestroy")
        super.onDestroy()
    }

    override fun onStart() {
        println("${this}: onStart")
        super.onStart()
    }

    override fun onRestart() {
        println("${this}: onRestart")
        super.onRestart()
    }

    override fun onStop() {
        println("${this}: onStop")
        super.onStop()
    }

    override fun onResume() {
        println("${this}: onResume")
        super.onResume()
    }

    override fun onPause() {
        println("${this}: onPause")
        super.onPause()
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
        println("${this}: onRestoreInstanceState")
        super.onRestoreInstanceState(savedInstanceState)
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        println("${this}: onSaveInstanceState")
        super.onSaveInstanceState(outState)
    }

}
